lib/css-tools.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+var csstree     = require('css-tree'),
+    stable      = require('stable'),
+    specificity = require('csso/lib/restructure/prepare/specificity');
+
+
+function flattenToSelectors(cssAst) {
+  var selectors   = [],
+      curSelector = null;
+
+  // TODO: walkup to avoid look-behind
+  csstree.walk(cssAst, function(node, item, list) {
+
+    // selector with pseudos
+    // look-behind: The pseudo selectors come after the affected simple selector
+    if(node.type == 'PseudoClassSelector' || 
+       node.type == 'PseudoElementSelector') {
+      curSelector.pseudos.push({ item: item, list: list });
+    }
+
+    // (simple) selector
+    if(node.type !== 'Selector') {
+      return;
+    }
+
+    curSelector = {
+      item:    item,
+      atrule:  this.atrule,
+      rule:    this.rule,
+      pseudos: []
+    };
+    selectors.push(curSelector);
+
+  });
+
+  return selectors;
+}
+
+
+function filterByMqs(selectors, useMqs) {
+  return selectors.filter(function(selector) {
+    if(selector.atrule === null) {
+      return useMqs.indexOf('') > -1;
+    }
+
+    var mqName = selector.atrule.name;
+    var mqExpr = csstree.translate(selector.atrule.expression);
+    var mqStr  = [mqName, mqExpr].join(' ');
+    console.log(mqStr);
+
+    return useMqs.indexOf(mqStr) > -1;
+  });
+}
+
+function filterByPseudos(selectors, usePseudos) {
+  var emptySelectorObj = {
+    type: 'Selector',
+    children: []
+  };
+  return selectors.filter(function(selector) {
+    var pseudoSelectorsAst = csstree.fromPlainObject(emptySelectorObj);
+    for(var pseudo of selector.pseudos) {
+      pseudoSelectorsAst.children.appendData(pseudo.item.data);
+    }
+
+    var pseudoSelectorsStr = csstree.translate(pseudoSelectorsAst);
+
+    return usePseudos.indexOf(pseudoSelectorsStr) > -1;
+  });
+}
+
+function cleanPseudos(selectors) {
+  selectors.map(function(selector) {
```

Comment:
As a CSS AST node only belongs to one parent node (and its list), it would be nice being able to access the list of an item directly (e.g. `item.list`) - currently with the item its containing list has to be kept for removal later.